### PR TITLE
[lua] Remove haxe-deps and directly reference dependencies; re-enable CI

### LINKED ISSUE #10777: Restoring lua tests
Seems like we forgot to fix the lua tests after "temporarily disabling them" in 8027d746888afb9aeeee077b737dec175417945a

We discussed this on slack a long time ago, but we need to update: https://github.com/jdonaldson/haxe-deps
as the `git:` protocol is no longer supported (which is used in the source url). I think it can just be updated to `https`.

@jdonaldson I made a PR for this, would you be able to handle making a release? https://github.com/jdonaldson/haxe-deps/pull/2

Also, it is worth considering adding `luasec` to the file, because it is now a potential dependency since: #10593. This haxe-deps project seems to be used to install all potential dependencies, even if they are not necessarily used in all haxe lua projects, so it makes sense to add it. Right now we have to install it manually for the tests: https://github.com/HaxeFoundation/haxe/blob/55f4b332022e6e84503db0f6bfa251112dcc8127/tests/runci/targets/Lua.hx#L73-L74

We should also probably mention haxe-deps in the manual, but that is an issue for that repository.

### PR DESCRIPTION
**Pull request by inklit**
This removes the abandoned haxe-deps LuaRocks reference for lua tests. 
While we could have forked it, it is more secure to just directly reference the dependencies (avoiding a supply chain attack, similar to those that have happened on nodejs package managers before). 

This resolves #10777 and #5024, and skips unit test Issue10752 (#10752) for Lua, and re-enables CI tests for Lua

### CODE REVIEW

**Comment by Simn** [comment]
Nice! And wow, they actually still pass, that's a nice surprise. In the long run, we should still sort out the haxe-deps situation, but for now this is a very welcome addition. Thank you!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -345,14 +345,14 @@ jobs:
       fail-fast: false
       matrix:
         # TODO enable lua after https://github.com/HaxeFoundation/haxe/issues/5024
```
Comment (danielo515) [inline .github/workflows/main.yml:347]: LOL, you could have remove this comment too 😂

**Comment by Uzume** [comment]
Thank you.

> [...]In the long run, we should still sort out the haxe-deps situation, but for now this is a very welcome addition.[...]

If we are to switch back to haxe-deps, I recommend we take it over and have that luarock be sourced from this repo (not a separate one). That said, I am not sure we really need something like haxe-deps even if it is sort of useful from a user perspective.

**Comment by danielo515** [comment]
What is Haxe-deps?

**Comment by Uzume** [comment]
> What is Haxe-deps?

@danielo515: [haxe-deps](https://luarocks.org/modules/jdonaldson/haxe-deps) or [jdonaldson/haxe-deps](https://github.com/jdonaldson/haxe-deps) is an apparently unmaintained LuaRocks package that contained nothing but dependencies necessary for Haxe.

At some point we switched to using that to ensure all the Lua dependencies for Haxe were installed (and it is handy for user installs), however, is has not been maintained/updated so now it is only deceptive.

I meant by my statement that if we go back to something like haxe-deps then it should be owned and maintained by this repo and not some random third-party. That way it can be kept up-to-date with the code that actually needs/uses it (since it lives in the same place).

**Pull request merged**